client/allocrunner/taskrunner/consul_hook.go

```diff
@@ -79,7 +79,8 @@ func (h *consulHook) Prestart(ctx context.Context, req *interfaces.TaskPrestartR
 			}
 
 			env := map[string]string{
-				"CONSUL_TOKEN": token.SecretID,
+				"CONSUL_TOKEN":      token.SecretID,
+				"CONSUL_HTTP_TOKEN": token.SecretID,
```

Comment:
Having a `template` block when Nomad is configured to use Workload Identity with Consul is what causes us to hit this code path in the first place. But the template hook runs after this hook so it will definitely override it if you've decided to, for example, fetch a time-limited Consul token from Vault.

This will overwrite a hard-coded `env` block field, but there's no recursive environment interpolation there, so it has to be a literal token string. Once you've got WI set up (again, that's what triggers this code path), there's no reason to do that and it's incredibly insecure to boot.

Reply:
(Also note that even if you could unset the env var, the token is written to `/secrets`, so it's not like this is a security issue where a job submitter can hide the token from the task.)